madlibs.js

```diff
@@ -1,42 +1,126 @@
+(function parseStory(rawStory) {
+  return fetch("story.txt")
+    .then((response) => response.text()) // A Promise that resolves with a String.
+    .then((data) =>story_to_obj(data.split('$$'))); // 3 story parts. split('$$') will separate text after $$ signs. Will create array with 3 elements. 
+})()
+
+// ==================|| modifiyng the story array, ?? ||===================
+function story_to_obj(story) {
+  for(let j=0;j<story.length;j++){ 
+    result = story[j].split(/(\.|, | )/g); // putting dots and commas and spaces as array's elements
+    console.log(result);
+    const noun = /\[n\]/; 
+    const verb = /\[v\]/;
+    const adj = /\[a\]/;
+    for (let i = 0; i < result.length; i++) {
+      if (noun.test(result[i])) {  // searching if noun exists in the iteration. If it exist it will return true.
+        id_index++; 
+        add_to_obj(i,'Noun',j);  // passing index into add_to_obj function as a parameter. j parameter is about paragraph number.
+      } else if (verb.test(result[i])) {
+        id_index++;
+        add_to_obj(i,'Verb',j);
+      } else if (adj.test(result[i])) {
+        id_index++;
+        add_to_obj(i,'Adjective',j);
+      } else {   
+        add_to_obj_word(i,j); 
+      }
+    }
+  }
+}
+
+// ==================|| modifiyng the story array, ?? ||===================
+function add_to_obj(index,pos,paragraph_index) {
+  ar[index] = {
+    word: ` <input type="text" id="input${id_index}" placeholder="${pos}" maxlength="20">`, //for the madlibsEdit part
+    word2: ` <input type="text" id="output${id_index}">`, //for the madlibsPreview part
+  };
+eval('paragraph'+paragraph_index).innerHTML += " " + ar[index].word;
+eval('previwe'+paragraph_index).innerHTML += " " + ar[index].word2;
+}
+
+function add_to_obj_word(index,paragraph_index) {
+  ar[index] = { word: `${result[index]}` }; // We are creating an object for each array index. ${result[index]} equals to the word of the paragraph, or elements of the array.
+  eval('paragraph'+paragraph_index).innerHTML += " " + ar[index].word; //  we are adding the word to the story dynamically through innerHTML. Variables created below.
+  eval('previwe'+paragraph_index).innerHTML += " " + ar[index].word; // the same for the preview part.
 }
 
-/**
- * All your other JavaScript code goes here, inside the function. Don't worry about
- * the `then` and `async` syntax for now.
- * 
- * You'll want to use the results of parseStory() to display the story on the page.
- */
-getRawStory().then(parseStory).then((processedStory) => {
-  console.log(processedStory);
+// ==================|| this part will be modified later ...  ||===================
+const madlibs0 = document.querySelector(".madLibsEdit0");
+const madlibs_pre0 = document.querySelector(".madLibsPreview0");
+const paragraph0 = document.createElement("p");
+const previwe0 = document.createElement("p");
+madlibs_pre0.append(previwe0);
```

Comment:
It would bee good to add some spaces and comments between this large part. Like what is madlibs1 and  madlibs2 just write a comment that distinguish things.